### PR TITLE
Adding Unit tests for Identity module

### PR DESCRIPTION
**Pull request by Ashi1993**
## Adding Unit tests for Identity module

> Explain  in a few lines the purpose of this pull request

**Issue link:** *required*

------

### Development Checklist

1. [ ] Built complete solution with pull request in place.
2. [ ] Ran checkstyle plugin with pull request in place.
3. [ ] Ran Findbugs plugin with pull request in place.
4. [ ] Formatted code according to WSO2 code style.

### Secure Development Checklist

1. [ ] Ran FindSecurityBugs plugin and verified report.
2. [ ] Ran Dependency-check plugin and verified report for new dependencies added.
2. [ ] Ran Dependency-check plugin and verified report for dependency version changes.
3. [ ] Have you verify the PR does't commit any keys, passwords, tokens, usernames, or other secrets?
4. [ ] Have you followed secure coding standards in [WSO2 Secure Engineering Guidelines](http://wso2.com/technical-reports/wso2-secure-engineering-guidelines)?

### Testing Checklist

1. [ ] Written unit tests.
2. [ ] Documented test scenarios(link available in guides).
3. [ ] Written automation tests (link available in guides).

**Automation Test Details**

| Test Suite        | Test Script IDs   |
| ----------------- | ----------------- |
| Integration Suite | *TCXXXXX, TCXXXX* |

### CODE REVIEW

**Review by anjuchamantha — DISMISSED** [review]

File: components/org.wso2.bfsi.identity.extensions/src/test/java/org/wso2/bfsi/identity/extensions/auth/extensions/request/validator/DefaultBFSIRequestObjectValidatorTest.java [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+        JOSEObject jwt = JOSEObject.parse(request);
+        if (jwt.getHeader().getAlgorithm() == null || jwt.getHeader().getAlgorithm().equals(JWSAlgorithm.NONE)) {
+            requestObject.setPlainJWT(PlainJWT.parse(request));
+        } else {
+            requestObject.setSignedJWT(SignedJWT.parse(request));
+        }
+        return new BFSIRequestObject<>(requestObject);
+    }
+
+}
```
Comment (anjuchamantha) [inline components/org.wso2.bfsi.identity.extensions/src/test/java/org/wso2/bfsi/identity/extensions/auth/extensions/request/validator/DefaultBFSIRequestObjectValidatorTest.java:125]:
```suggestion
}

```

File: components/org.wso2.bfsi.identity.extensions/src/test/java/org/wso2/bfsi/identity/extensions/auth/extensions/request/validator/ReqObjectTestDataProvider.java [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    @DataProvider(name = "dp-checkInValidRequestObject")
+    public Object[][] dpCheckInValidRequestObject() {
+
+        return new Object[][]{
+                {null, new OAuth2Parameters()},
+                {null, null},
+        };
+    }
+
+}
```
Comment (anjuchamantha) [inline components/org.wso2.bfsi.identity.extensions/src/test/java/org/wso2/bfsi/identity/extensions/auth/extensions/request/validator/ReqObjectTestDataProvider.java:133]:
```suggestion
}

```

File: components/org.wso2.bfsi.identity.extensions/src/test/java/org/wso2/bfsi/identity/extensions/auth/extensions/request/validator/RequestObjectValidatorTest.java [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+        // Assert
+        assertEquals(3, ukRequestObject.getRequestedClaims().size());
+
+        // Assert
+        assertEquals("code id_token", ukRequestObject.getClaim("response_type"));
+
+        // Assert
+        assertEquals("code id_token", ukRequestObject.getClaimValue("response_type"));
+    }
+}
```
Comment (anjuchamantha) [inline components/org.wso2.bfsi.identity.extensions/src/test/java/org/wso2/bfsi/identity/extensions/auth/extensions/request/validator/RequestObjectValidatorTest.java:139]:
```suggestion
}

```

File: components/org.wso2.bfsi.identity.extensions/src/test/java/org/wso2/bfsi/identity/extensions/auth/extensions/response/handler/ResponseTypeHandlerTest.java [hunk 4]
```diff
@@ -0,0 +1,360 @@
...
+        when(mock.getAuthorizationReqDTO()).thenReturn(mockAuthReq);
+        when(mockAuthReq.getSessionDataKey()).thenReturn("123");
+
+        BFSIResponseTypeHandler uut = new BFSIDefaultResponseTypeHandlerImpl();
+        String[] result = uut.updateApprovedScopes(mock);
+
+        assertEquals(1, result.length);
+
+    }
+}
```
Comment (anjuchamantha) [inline components/org.wso2.bfsi.identity.extensions/src/test/java/org/wso2/bfsi/identity/extensions/auth/extensions/response/handler/ResponseTypeHandlerTest.java:360]:
```suggestion
}

```

File: components/org.wso2.bfsi.identity.extensions/src/test/java/org/wso2/bfsi/identity/extensions/auth/extensions/response/validator/OBCodeResponseTypeValidatorTest.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+            mock.when(() -> CommonUtils.isRegulatoryApp(anyString())).thenReturn(false);
+
+            BFSICodeResponseTypeValidator uut = spy(new BFSICodeResponseTypeValidator());
+
+            // Act
+            uut.validateRequiredParameters(httpServletRequestMock);
+        }
+    }
+}
```
Comment (anjuchamantha) [inline components/org.wso2.bfsi.identity.extensions/src/test/java/org/wso2/bfsi/identity/extensions/auth/extensions/response/validator/OBCodeResponseTypeValidatorTest.java:57]:
```suggestion
}

```

File: components/org.wso2.bfsi.identity.extensions/src/test/java/org/wso2/bfsi/identity/extensions/auth/extensions/response/validator/OBHybridResponseTypeValidatorTest.java [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+        // Mock
+        HttpServletRequest httpServletRequestMock = mock(HttpServletRequest.class);
+        when(httpServletRequestMock.getMethod()).thenReturn("PUT");
+
+        BFSIHybridResponseTypeValidator uut = spy(new BFSIHybridResponseTypeValidator());
+
+        // Act
+        uut.validateMethod(httpServletRequestMock);
+    }
+}
```
Comment (anjuchamantha) [inline components/org.wso2.bfsi.identity.extensions/src/test/java/org/wso2/bfsi/identity/extensions/auth/extensions/response/validator/OBHybridResponseTypeValidatorTest.java:113]:
```suggestion
}

```

File: components/org.wso2.bfsi.identity.extensions/src/test/java/org/wso2/bfsi/identity/extensions/claims/BFSIDefaultClaimProviderTest.java [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+                doReturn(authorizeReqDTO).when(authzReqMessageContext).getAuthorizationReqDTO();
+
+                Map<String, Object> claims = bfsiDefaultClaimProvider
+                        .getAdditionalClaims(authzReqMessageContext, authorizeRespDTO);
+
+                assertFalse(claims.isEmpty());
+            }
+        }
+    }
+}
```
Comment (anjuchamantha) [inline components/org.wso2.bfsi.identity.extensions/src/test/java/org/wso2/bfsi/identity/extensions/claims/BFSIDefaultClaimProviderTest.java:87]:
```suggestion
}

```

File: components/org.wso2.bfsi.identity.extensions/src/test/java/org/wso2/bfsi/identity/extensions/claims/BFSIDefaultOIDCClaimsCallbackHandlerTest.java [hunk 8]
```diff
@@ -0,0 +1,164 @@
...
+            assertEquals("123", jwtClaimsSet.getClaim("consent_id"));
+            assertTrue(jwtClaimsSet.getClaim("cnf").toString().contains("{x5t#S256=LS0tLS1CRUdJTiBDRVJUSUZJQ0F" +
+                    "URS0tLS0tCk1JSUZPRENDQkNDZ0F3SUJBZ0lFV2NiaWlUQU5CZ2txaGtpRzl3MEJBUXNGQURCVE1Rc3dDUVlEVlFRR0V3Sk" +
+                    "gKUWpFVU1CSUdBMVVFQ2hNTFQzQmxia0poYm10cGJtY3hMakFzQmdOVkJBTVRKVTl3Wlc1Q1lXNXJhVzVuSUZCeQpaUzFRY" +
+                    "205a2RXTjBhVzl1SUVsemMzVnBibWNnUTBFd0hoY05Nak14TVRFMU1EVXhNRE14V2hjTk1qUXhNakUxCk1EVTBNRE14V2pC" +
+                    "aE1Rc3dDUVlEV"));
+            assertEquals("aaa@gold.com", jwtClaimsSet.getClaim("sub"));
+        }
+    }
+}
```
Comment (anjuchamantha) [inline components/org.wso2.bfsi.identity.extensions/src/test/java/org/wso2/bfsi/identity/extensions/claims/BFSIDefaultOIDCClaimsCallbackHandlerTest.java:164]:
```suggestion
}

```

File: components/org.wso2.bfsi.identity.extensions/src/test/java/org/wso2/bfsi/identity/extensions/claims/RoleClaimProviderImplTest.java [hunk 9]
```diff
@@ -0,0 +1,185 @@
...
+            Mockito.when(oAuthTokenReqMessageContextMock.getScope()).thenReturn(SCOPES);
+            Mockito.when(oAuthTokenReqMessageContextMock.getAuthorizedUser()).thenReturn(authorizedUserMock);
+
+            Map<String, Object> claims = uut.getAdditionalClaims(oAuthTokenReqMessageContextMock, null);
+
+            // assert
+            Assert.assertFalse(claims.containsKey("user_role"));
+        }
+    }
+}
```
Comment (anjuchamantha) [inline components/org.wso2.bfsi.identity.extensions/src/test/java/org/wso2/bfsi/identity/extensions/claims/RoleClaimProviderImplTest.java:185]:
```suggestion
}

```

File: components/org.wso2.bfsi.identity.extensions/src/test/java/org/wso2/bfsi/identity/extensions/util/IdentityCommonUtilsTest.java [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+     */
+    public static boolean hasExpired(X509Certificate peerCertificate) {
+        try {
+            peerCertificate.checkValidity();
+        } catch (CertificateException e) {
+            return true;
+        }
+        return false;
+    }
+}
```
Comment (anjuchamantha) [inline components/org.wso2.bfsi.identity.extensions/src/test/java/org/wso2/bfsi/identity/extensions/util/IdentityCommonUtilsTest.java:94]: Add new line, check other places as well.

File: components/org.wso2.bfsi.identity.extensions/src/test/java/org/wso2/bfsi/identity/extensions/validator/resources/ValidatorTestDataProvider.java [hunk 11]
```diff
@@ -0,0 +1,157 @@
...
+                {claimsTemplate.replace("${SCOPE}",
+                        "openid bank:payees:read")},
+                {claimsTemplate.replace("${SCOPE}",
+                        "openid bank:regular_payments:read")},
+                {claimsTemplate.replace("${SCOPE}",
+                        "openid common:customer.basic:read")},
+                {claimsTemplate.replace("${SCOPE}",
+                        "openid common:customer.detail:read")},
+
+                //
```
Comment (RivinduM) [inline components/org.wso2.bfsi.identity.extensions/src/test/java/org/wso2/bfsi/identity/extensions/validator/resources/ValidatorTestDataProvider.java]:
```suggestion
```

File: components/org.wso2.bfsi.identity.extensions/src/test/java/org/wso2/bfsi/identity/extensions/validator/resources/ValidatorTestDataProvider.java [hunk 12]
```diff
@@ -0,0 +1,157 @@
...
+                {claimsTemplate.replace("${SCOPE}",
+                        "accounts openid")},
+                {claimsTemplate.replace("${SCOPE}",
+                        "openid payments")},
+                {claimsTemplate.replace("${SCOPE}",
+                        "openid fundsconfirmations")},
+
+                //
```
Comment (RivinduM) [inline components/org.wso2.bfsi.identity.extensions/src/test/java/org/wso2/bfsi/identity/extensions/validator/resources/ValidatorTestDataProvider.java]:
```suggestion
```

File: components/org.wso2.bfsi.identity.extensions/src/test/java/org/wso2/bfsi/identity/extensions/validator/resources/ValidatorTestDataProvider.java [hunk 13]
```diff
@@ -0,0 +1,157 @@
...
+                {claimsTemplate.replace("${SCOPE}",
+                        "openid xyz")},
+                {claimsTemplate.replace("${SCOPE}",
+                        "xyz")},
+                {claimsTemplate.replace("${SCOPE}",
+                        "openid bank:accounts.basic")},
+                {claimsTemplate.replace("${SCOPE}",
+                        "openid common:customer.detail:read")},
+
+                //
```
Comment (RivinduM) [inline components/org.wso2.bfsi.identity.extensions/src/test/java/org/wso2/bfsi/identity/extensions/validator/resources/ValidatorTestDataProvider.java]:
```suggestion
```

File: components/org.wso2.bfsi.identity.extensions/src/test/java/org/wso2/bfsi/identity/extensions/validator/resources/ValidatorTestDataProvider.java [hunk 14]
```diff
@@ -0,0 +1,157 @@
...
+                {claimsTemplate.replace("${SCOPE}",
+                        "Accounts openid")},
+                {claimsTemplate.replace("${SCOPE}",
+                        "openid Payments")},
+                {claimsTemplate.replace("${SCOPE}",
+                        "openid FundsConfirmations")},
+
+                //
```
Comment (RivinduM) [inline components/org.wso2.bfsi.identity.extensions/src/test/java/org/wso2/bfsi/identity/extensions/validator/resources/ValidatorTestDataProvider.java]:
```suggestion
```

File: components/org.wso2.bfsi.identity.extensions/src/test/java/org/wso2/bfsi/identity/extensions/validator/ModelValidatorsTest.java [hunk 15]
```diff
@@ -0,0 +1,131 @@
...
+
+        //
+        sampleRequestObject = new SampleDifferentClass();
+        sampleRequestObject.setName("name");
+        sampleRequestObject.setAge(70);
+
+        violation = BFSIValidator.getInstance().getFirstViolation(sampleRequestObject);
+        assertEquals("male failed", violation);
+
+        //
```
Comment (RivinduM) [inline components/org.wso2.bfsi.identity.extensions/src/test/java/org/wso2/bfsi/identity/extensions/validator/ModelValidatorsTest.java]:
```suggestion
```

File: components/org.wso2.bfsi.identity.extensions/src/test/java/org/wso2/bfsi/identity/extensions/validator/ModelValidatorsTest.java [hunk 16]
```diff
@@ -0,0 +1,131 @@
...
+    public void checkOpenBankingValidator() {
+
+        SampleDifferentClass sampleRequestObject = new SampleDifferentClass();
+        sampleRequestObject.setName("name");
+        sampleRequestObject.setMale(true);
+
+        String violation = BFSIValidator.getInstance().getFirstViolation(sampleRequestObject);
+        assertEquals("age failed", violation);
+
+        //
```
Comment (RivinduM) [inline components/org.wso2.bfsi.identity.extensions/src/test/java/org/wso2/bfsi/identity/extensions/validator/ModelValidatorsTest.java]:
```suggestion
```

File: components/org.wso2.bfsi.identity.extensions/src/test/java/org/wso2/bfsi/identity/extensions/validator/ModelValidatorsTest.java [hunk 17]
```diff
@@ -0,0 +1,131 @@
...
+
+        SampleDifferentClass sampleRequestObject = new SampleDifferentClass();
+        sampleRequestObject.setName("name");
+        sampleRequestObject.setMale(true);
+
+        Set<ConstraintViolation<SampleDifferentClass>> violations = validator.validate(sampleRequestObject);
+        String violation = violations.stream().findFirst().map(ConstraintViolation::getMessage).orElse(null);
+        assertEquals("age failed", violation);
+
+        //
```
Comment (RivinduM) [inline components/org.wso2.bfsi.identity.extensions/src/test/java/org/wso2/bfsi/identity/extensions/validator/ModelValidatorsTest.java]: Shall we add a proper comment or remove this line?

File: components/org.wso2.bfsi.identity.extensions/src/test/java/org/wso2/bfsi/identity/extensions/validator/ModelValidatorsTest.java [hunk 18]
```diff
@@ -0,0 +1,131 @@
...
+        sampleRequestObject = new SampleDifferentClass();
+        sampleRequestObject.setName("name");
+        sampleRequestObject.setAge(70);
+
+        violations = validator2.validate(sampleRequestObject);
+        violation = violations.stream().findFirst().map(ConstraintViolation::getMessage).orElse(null);
+        assertEquals("male failed", violation);
+
+        //
```
Comment (RivinduM) [inline components/org.wso2.bfsi.identity.extensions/src/test/java/org/wso2/bfsi/identity/extensions/validator/ModelValidatorsTest.java]:
```suggestion
```

Comment (Ashi1993) [reply components/org.wso2.bfsi.identity.extensions/src/test/java/org/wso2/bfsi/identity/extensions/validator/ModelValidatorsTest.java]: Fixed in https://github.com/wso2-extensions/bfsi-consent-management/pull/13/commits/cd47ead8b0ab89e7914625db1aa8437e57dc4655

File: components/org.wso2.bfsi.identity.extensions/src/test/java/org/wso2/bfsi/identity/extensions/validator/ModelValidatorsTest.java (hunk 16 above)
Comment (Ashi1993) [reply components/org.wso2.bfsi.identity.extensions/src/test/java/org/wso2/bfsi/identity/extensions/validator/ModelValidatorsTest.java]: Fixed in https://github.com/wso2-extensions/bfsi-consent-management/pull/13/commits/cd47ead8b0ab89e7914625db1aa8437e57dc4655

File: components/org.wso2.bfsi.identity.extensions/src/test/java/org/wso2/bfsi/identity/extensions/validator/ModelValidatorsTest.java (hunk 15 above)
Comment (Ashi1993) [reply components/org.wso2.bfsi.identity.extensions/src/test/java/org/wso2/bfsi/identity/extensions/validator/ModelValidatorsTest.java]: Fixed in https://github.com/wso2-extensions/bfsi-consent-management/pull/13/commits/cd47ead8b0ab89e7914625db1aa8437e57dc4655

File: components/org.wso2.bfsi.identity.extensions/src/test/java/org/wso2/bfsi/identity/extensions/validator/resources/ValidatorTestDataProvider.java (hunk 14 above)
Comment (Ashi1993) [reply components/org.wso2.bfsi.identity.extensions/src/test/java/org/wso2/bfsi/identity/extensions/validator/resources/ValidatorTestDataProvider.java]: Fixed in https://github.com/wso2-extensions/bfsi-consent-management/pull/13/commits/cd47ead8b0ab89e7914625db1aa8437e57dc4655

File: components/org.wso2.bfsi.identity.extensions/src/test/java/org/wso2/bfsi/identity/extensions/validator/resources/ValidatorTestDataProvider.java (hunk 13 above)
Comment (Ashi1993) [reply components/org.wso2.bfsi.identity.extensions/src/test/java/org/wso2/bfsi/identity/extensions/validator/resources/ValidatorTestDataProvider.java]: Fixed in https://github.com/wso2-extensions/bfsi-consent-management/pull/13/commits/cd47ead8b0ab89e7914625db1aa8437e57dc4655

File: components/org.wso2.bfsi.identity.extensions/src/test/java/org/wso2/bfsi/identity/extensions/validator/resources/ValidatorTestDataProvider.java (hunk 12 above)
Comment (Ashi1993) [reply components/org.wso2.bfsi.identity.extensions/src/test/java/org/wso2/bfsi/identity/extensions/validator/resources/ValidatorTestDataProvider.java]: Fixed in https://github.com/wso2-extensions/bfsi-consent-management/pull/13/commits/cd47ead8b0ab89e7914625db1aa8437e57dc4655

File: components/org.wso2.bfsi.identity.extensions/src/test/java/org/wso2/bfsi/identity/extensions/validator/resources/ValidatorTestDataProvider.java (hunk 11 above)
Comment (Ashi1993) [reply components/org.wso2.bfsi.identity.extensions/src/test/java/org/wso2/bfsi/identity/extensions/validator/resources/ValidatorTestDataProvider.java]: Fixed in https://github.com/wso2-extensions/bfsi-consent-management/pull/13/commits/cd47ead8b0ab89e7914625db1aa8437e57dc4655

File: components/org.wso2.bfsi.identity.extensions/src/test/java/org/wso2/bfsi/identity/extensions/util/IdentityCommonUtilsTest.java (hunk 10 above)
Comment (Ashi1993) [reply components/org.wso2.bfsi.identity.extensions/src/test/java/org/wso2/bfsi/identity/extensions/util/IdentityCommonUtilsTest.java:94]: Line is already there, hence resolving comments

**Review by RivinduM — APPROVED** [review]

**Review by anjuchamantha — APPROVED** [review]

**Pull request merged**